kernel/src/process_load_utilities.rs

```diff
@@ -0,0 +1,432 @@
+use crate::syscall_driver::{CommandReturn, SyscallDriver};
+use crate::process::ProcessId;
+use crate::ErrorCode;
+
+pub const DRIVER_NUM: usize = 0x10001;
+
+//FLASH_START_ADDRESS and UNUSED_RAM_START_ADDRESS are used in order to load an application flashed from OTA_app
+static mut FLASH_START_ADDRESS: usize = 0x00040000;
+pub static mut UNUSED_RAM_START_ADDRESS: usize = 0x20006000;
+pub static mut INDEX_OF_PROCESSES: usize = 0;
+
+mod ro_allow {
+    pub(crate) const WRITE: usize = 0;
+    /// The number of allow buffers the kernel stores for this grant
+    pub(crate) const COUNT: usize = 1;
+}
+
+/// Ids for read-write allow buffers
+mod rw_allow {
+    pub(crate) const READ: usize = 0;
+    /// The number of allow buffers the kernel stores for this grant
+    pub(crate) const COUNT: usize = 1;
+}
+
+/// State that is stored in each process's grant region to support process_load_utilities.
+#[derive(Default)]
+struct ProcLoaderData;
+
+pub struct ProcessLoader <C:'static + Chip>{
+    kernel: &'static Kernel,
+    chip: &'static C, 
+    fault_policy: &'static dyn ProcessFaultPolicy,
+    ptr_process: *mut Option<&'static (dyn Process + 'static)>,
+    process_num: usize,
+    data: Grant<
+        ProcLoaderData,
+        UpcallCount<2>,
+        AllowRoCount<{ ro_allow::COUNT }>,
+        AllowRwCount<{ rw_allow::COUNT }>,
+    >,
+}
+
+impl <C:'static + Chip> ProcessLoader <C> {
+    pub fn init(
+        kernel: &'static Kernel,
+        chip: &'static C,
+        fault_policy: &'static dyn ProcessFaultPolicy,
+        memcapability: &dyn MemoryAllocationCapability,
+        ptr_process: *mut Option<&'static (dyn Process + 'static)>,
+        process_num: usize,
+    ) -> ProcessLoader <C> {
+        ProcessLoader {
+            kernel: kernel,
+            chip: chip, 
+            fault_policy: fault_policy,
+            ptr_process: ptr_process,
+            process_num: process_num,
+            data: kernel.create_grant(DRIVER_NUM, memcapability),
+        }
+    }
+
+    // load_processes_advanced_air is implemented based on load_processes_advanced
+    // the purpose of this function is to load an application (process) flashed from OTA_app
+    fn load_processes_advanced_air(&self) -> Result<(), ProcessLoadError> { 
+        // These symbols are defined in the linker script.
+        extern "C" {
+            /// Beginning of the ROM region containing app images.
+            static mut _sapps: u8;
+            /// End of the ROM region containing app images.
+            static _eapps: u8;
+            /// Beginning of the RAM region for app memory.
+            static mut _sappmem: u8;
+            /// End of the RAM region for app memory.
+            static _eappmem: u8;
+        }
+    
+        unsafe{
+            debug!("flash start address {:#010X}", FLASH_START_ADDRESS);
+            debug!("ram start address {:#010X}", UNUSED_RAM_START_ADDRESS);
+            debug!("Index {:?}", INDEX_OF_PROCESSES);
+        };
```

Comment:
This shouldn't be printing by default. You could probably tie it in with `debug_load_processes` though

Reply:
I have deleted that test code